Frends.Zip.CreateArchive/Frends.Zip.CreateArchive.Tests/UnitTests.cs

```diff
@@ -5,316 +5,288 @@
+        File.WriteAllText(Path.Combine(_subDirIn, dublicateFileName), "Seaman: Swallow, come!");
+        var subDir2 = Path.Combine(_subDirIn, "subdir2");
+        Directory.CreateDirectory(subDir2);
+        File.WriteAllText(Path.Combine(subDir2, dublicateFileName), "Seaman: Swallow, come!");
+
+        _source.IncludeSubFolders = true;
+        _source.FlattenFolders = true;
+        _destination.RenameDuplicateFiles = true;
+
+        var result = ExecuteCreateArchive();
+        Assert.AreEqual(_zipFileName, Path.GetFileName(result.Path));
+        Assert.AreEqual(7, result.FileCount);
+        var subDirCount = result.ArchivedFiles.Where(s => s.Contains(_subDir)).Count();
+        Assert.AreEqual(0, subDirCount);
+        Assert.Contains("dublicate_file.txt", result.ArchivedFiles);
+        Assert.Contains("dublicate_file_(1).txt", result.ArchivedFiles);
+        Assert.Contains("dublicate_file_(2).txt", result.ArchivedFiles);
+    }
 
-            ExecuteCreateArchive();
-            ExecuteCreateArchive();
+    [Test]
+    public void ZipFiles_FileList_FlattensFolders()
+    {
+        _source.FlattenFolders = false;
+        _source.Directory = "";
+        _source.SourceType = SourceFilesType.FileList;
+        var filePath = Directory.GetFiles(_dirIn)[0];
+        _source.FilePathsList = new List<string> { filePath };
+        var result = ExecuteCreateArchive();
+        Assert.AreEqual(Path.GetFileName(filePath), result.ArchivedFiles[0]);
+    }
 
-            Assert.AreEqual(1, Directory.GetFiles(_destination.Directory, "*.zip").Count());
-        }
+    [Test]
+    public void ZipFile_Exists_ThrowsError()
+    {
+        _options.DestinationFileExistsAction = FileExistAction.Error;
+        ExecuteCreateArchive();
+        var result = Assert.Throws<Exception>(() => ExecuteCreateArchive());
+        Assert.IsTrue(result.Message.Equals($"Destination file {Path.Combine(_destination.Directory, _destination.FileName)} already exists."));
+    }
 
-        [Test]
-        public void ZipFile_Exists_Rename()
-        {
-            _options.DestinationFileExistsAction = FileExistAction.Rename;
+    [Test]
+    public void ZipFile_Exists_OverwriteFile()
+    {
+        _options.DestinationFileExistsAction = FileExistAction.Overwrite;
+        ExecuteCreateArchive();
+        ExecuteCreateArchive();
+        Assert.AreEqual(1, Directory.GetFiles(_destination.Directory, "*.zip").Count());
+    }
 
-            var result1 = ExecuteCreateArchive();
-            var result2 = ExecuteCreateArchive();
-            var result3 = ExecuteCreateArchive();
+    [Test]
+    public void ZipFile_Exists_Rename()
+    {
+        _options.DestinationFileExistsAction = FileExistAction.Rename;
+        var result1 = ExecuteCreateArchive();
+        var result2 = ExecuteCreateArchive();
+        var result3 = ExecuteCreateArchive();
+        var zipFiles = Directory.GetFiles(_destination.Directory, "*.zip");
+        Assert.AreEqual(3, zipFiles.Count());
```

Comment:
Variable `zipFiles` is an array, not a List, so you should use `.Length` instead of `.Count`.